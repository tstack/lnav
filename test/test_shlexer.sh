#! /bin/bash

export FOO='bar'
export DEF='xyz'

run_cap_test ./drive_shlexer '$FOO'

run_cap_test ./drive_shlexer '${FOO}'

# run_cap_test ./drive_shlexer '\a'

run_cap_test ./drive_shlexer '\'

run_cap_test ./drive_shlexer "'abc'"

run_cap_test ./drive_shlexer '"def"'

run_cap_test ./drive_shlexer '"'"'"'"'

run_cap_test ./drive_shlexer "'"'"'"'"

run_cap_test ./drive_shlexer '"abc $DEF 123"'

run_cap_test ./drive_shlexer '"abc ${DEF} 123"'

run_cap_test ./drive_shlexer "'abc \$DEF 123'"

run_cap_test ./drive_shlexer 'abc $DEF  123'

run_cap_test ./drive_shlexer '~ foo'

run_cap_test ./drive_shlexer '~nonexistent/bar baz'

run_cap_test ./drive_shlexer 'abc '
