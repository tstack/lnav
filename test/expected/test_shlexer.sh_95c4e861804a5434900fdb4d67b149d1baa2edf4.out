    $FOO
ref ^--^
eof    ^
eval -- bar
split:
  0 ^--^ -- bar
